pkg/writer/serializer_cdx14.go

```diff
@@ -182,43 +200,80 @@ func nodeToCDX14Component(n *sbom.Node) *cdx14.Component {
 	}
 
 	if n.Licenses != nil && len(n.Licenses) > 0 {
-		c.Licenses = []cdx14.License{}
+		var licenseChoices []cdx14.LicenseChoice
+		var licenses cdx14.Licenses
 		for _, l := range n.Licenses {
-			c.Licenses = append(c.Licenses, cdx14.License{
-				License: struct {
-					ID string "json:\"id\"" // TODO optimize
-				}{l},
+			licenseChoices = append(licenseChoices, cdx14.LicenseChoice{
+				License: &cdx14.License{
+					ID: l,
+				},
 			})
 		}
+
+		licenses = licenseChoices
+		c.Licenses = &licenses
 	}
 
 	if n.Hashes != nil && len(n.Hashes) > 0 {
-		c.Hashes = []cdx14.Hash{}
+		c.Hashes = &[]cdx14.Hash{}
 		for algo, hash := range n.Hashes {
-			c.Hashes = append(c.Hashes, cdx14.Hash{
-				Algorithm: algo, // Fix to make it valid
-				Content:   hash,
+			*c.Hashes = append(*c.Hashes, cdx14.Hash{
+				Algorithm: NormalizeAlgo(algo), // Fix to make it valid
+				Value:     hash,
 			})
 		}
 	}
 
 	if n.ExternalReferences != nil {
 		for _, er := range n.ExternalReferences {
 			if er.Type == "purl" {
-				c.Purl = er.Url
+				c.PackageURL = er.Url
 				continue
 			}
 
 			if c.ExternalReferences == nil {
-				c.ExternalReferences = []cdx14.ExternalReference{}
+				c.ExternalReferences = &[]cdx14.ExternalReference{}
 			}
 
-			c.ExternalReferences = append(c.ExternalReferences, cdx14.ExternalReference{
-				Type: er.Type,
+			*c.ExternalReferences = append(*c.ExternalReferences, cdx14.ExternalReference{
+				Type: cdx14.ExternalReferenceType(er.Type), // Fix to make it valid
 				URL:  er.Url,
 			})
 		}
 	}
 
 	return c
 }
+
+func NormalizeAlgo(algo string) cdx14.HashAlgorithm {
+	normAlgo := strings.ReplaceAll(strings.ToLower(algo), "_", "-")
+	switch normAlgo {
+	case "md5", "md-5":
```

Comment:
I think we should adopt a single const for each of the algos in the format package. We should never get "sha1" or  "sha-1" in a protobom. The readers should normalize to predefined names

Reply:
The .proto will have these kinds of fields, right?  So the reader will not really be able to choose what to call anything - the protobuf definition will have predefined field names and the readers will have to use those.

Reply:
Ah you are right, we should enum them in the proto :+1:

Reply:
Added the enum, but it exposed something interesting (some algos are only supported in SPDX). 

```
enum HashAlgorithm {
    UNKNOWN = 0;
    MD5 = 1;
    SHA1 = 2;
    SHA256 = 3;
    SHA384 = 4;
    SHA512 = 5;
    SHA3_256 = 6;
    SHA3_384 = 7;
    SHA3_512 = 8;
    BLAKE2B_256 = 9;
    BLAKE2B_384 = 10;
    BLAKE2B_512 = 11;
    BLAKE3 = 12;

    //2DO what should we do
    MD2 = 13; // Only supported by SPDX
    ADLER32 = 14; // Only supported by SPDX
    MD4 = 15; // Only supported by SPDX
    MD6 = 16; // Only supported by SPDX
    SHA224 = 17; // Only supported by SPDX
  }
  ```

Reply:
Yep, this will result in data loss, another one to document.